druid/src/widget/flex.rs

```diff
@@ -942,6 +1056,99 @@ impl<T: Data> Widget<T> for Flex<T> {
             ..Default::default()
         }
     }
+
+    fn compute_max_intrinsic_width(
+        &mut self,
+        ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &T,
+        env: &Env,
+    ) -> f64 {
+        match self.direction {
+            Axis::Horizontal => {
+                let f = |widget: &mut WidgetPod<T, Box<dyn Widget<T>>>| {
+                    widget
+                        .widget_mut()
+                        .compute_max_intrinsic_width(ctx, bc, data, env)
```

Comment:
Have you considered introducing a `Widget::compute_max_intrinsic(axis: Axis, ctx, bc, data, env)` method? It seems like it might get rid of some duplication here and maybe even eliminate the closure parameters to `measure_main_intrinsic` and `measure_cross_intrinsic`?

Reply:
Hmmm. `Axis` makes sense only for flex.

Reply:
Flex is definitely the big one, but I think there are others. In `Container` and `Padding`, `compute_max_intrinsic_height` is basically a copy-paste of `compute_max_intrinsic_width`, and I'd sort of expect that to be the most common case, would you not?

Reply:
It isnt a copy-paste. One computes height while other computes width. They _look_ the same. My biggest problem when I impl this was that those computes are converse of each other. I guess that's what you are alluding to?
Biggest pain was flex. I could have separated them out (not have closures) but it'd have been a nightmare maintenance wise. For simpler things like `Padding` I choose not to use closures because too much abstraction is difficult to follow.

Reply:
I still feel maybe Im missing your point. lol. Sorry. .If I did please elaborate or create a gist.

Reply:
What I meant was: if you had only one function but it had an `axis: Axis` parameter, then the implementation for `Padding` would just be the following function, with no need to have two functions that look almost the same.
```rust
fn compute_max_intrinsic(
        &mut self,
        axis: Axis,
        ctx: &mut LayoutCtx,
        bc: &BoxConstraints,
        data: &T,
        env: &Env,
    ) -> f64 {
        let inset_size = self.insets.resolve(env).size();
        let child_bc = bc.shrink(inset_size);
        let child_max_intrinsic = self
            .child
            .widget_mut()
            .compute_max_intrinsic(axis, ctx, &child_bc, data, env);
        child_max_intrinsic + axis.major(inset_size)
    }
```

Reply:
Why always `axis.major(size)` at the end? FYI Im computing not just max width of a `Flex::row` or max height of a `Flex::column`. Im also computing max height of `Flex::row` and max width of a `Flex::column`. When sizing and direction are opposite, you'd need `axis.cross(size)`. No?

Reply:
The example above was for `Padding`, not `Flex`.

Yes, `Flex` would be more complicated and involve the cross-axis also. Still, I think passing the axis and using `major` and `cross` would be easier to follow than passing two callbacks.

Reply:
No, no: the axis parameter specifies whether we're interested in the max intrinsic width or the max intrinsic height. It's not related to the axis of the Flex. I'm proposing to replace
```rust
fn compute_max_intrinsic_width(&mut self, ...) -> f64;
fn compute_max_intrinsic_height(&mut self, ...) -> f64;
```
by
```rust
fn compute_max_intrinsic(&mut self, Axis::Horizontal, ...) -> f64;
fn compute_max_intrinsic(&mut self, Axis::Vertical, ...) -> f64;
```
because for many widgets I suspect that `compute_max_intrinsic_width` and `compute_max_intrinsic_height` will mostly be copy-pastes of one another.

Reply:
OK I'll give this a try.

Reply:
I'm glad it worked out!

Reply:
Added another commit.